### DISCUSSION TITLE
how to publish binary data

### DISCUSSION THREAD

**Discussion by Nazneenashrafi** [emqx/MQTTX#1678] [Q&A]
warm hello to all
I want to publish binary data by MQTTX application how to do that ....
please help!!!!!!!!!!!!!!!

**Comment by ysfscream** [comment]
This feature is not directly supported in the current version of the desktop application but is on the development roadmap for the MQTTX CLI.

<img width="721" alt="image" src="https://github.com/emqx/MQTTX/assets/21299158/1e266343-e949-43bf-a760-b2776f952fc6">

However, you should know that MQTT data transmission is ultimately based on binary. You can convert your binary data into another format like hex or base64, which will be transformed back into binary during data publishing. On the receiving side, you can handle this binary data appropriately.

<img width="1095" alt="image" src="https://github.com/emqx/MQTTX/assets/21299158/172a3ff9-adce-47aa-aed6-3d39d78bfc05">
